pkg/clientcmd/configutil.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clientcmd
+
+import (
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+)
+
+// This file contains convenience methods for working with the kubeconfig and loading specific configurations from it
+
+// Creates an api.Config by searching for the kubeconfig on the default search path. If an override 'kubeconfigPath' is
+// given then that path is used instead of the default path.
+func NewApiConfigForDefaultSearchPath(kubeconfigPath string) (*api.Config, error) {
+	configLoader := clientcmd.NewDefaultClientConfigLoadingRules()
+	configLoader.ExplicitPath = kubeconfigPath
+	return configLoader.Load()
+}
+
+// Creates an api.Config for a given kubeconfig string.
+func NewApiConfigForKubeconfig(kubeconfig string) (*api.Config, error) {
+	return clientcmd.Load([]byte(kubeconfig))
```

Comment:
Does this need any extra error checking / handling since it's exposed outside of this package?

Reply:
The reason to do so would be to make this easier to use -- I can maybe check for empty string but i'm not sure about the value. the Load method being called would return an error in that case.